test/test_model.py

```diff
@@ -1355,15 +1393,18 @@ def setUp(self):
               bar:
                 k: v
         ''')
-        self.peer_rel_id = self.harness.add_relation('db2', 'db2')
-        self.app = self.harness.model.app
-        self.addCleanup(self.harness.cleanup)
+        yield harness
+        harness.cleanup()
+
+    @pytest.fixture
+    def app(self, harness: ops.testing.Harness[ops.CharmBase]):
+        return harness.model.app
```

Comment:
Similar to earlier comments, I think we should remove this fixture. We can just use a local variable `app = harness.model.app` if we want to avoid repeated `harness.model.app` expressions.